84-sarcastic-fringehead/technical_specification.md

```diff
@@ -0,0 +1,478 @@
+- [ ] Move migrations to own CLI command so they can be run as an init container
+  - [ ] Work with DevOps to get this configured in k8s
+
+### DINS:
+The DINS subscribes to `FileInternallyRegistered` events from the IFRS to learn about which files are available in the Archive. It also hosts an HTTP API where the outside world can obtain basic information about those same files.
+
+#### Work to be performed for the DINS
+- [ ] Get the updated `ghga-event-schemas` version and adapt DINS for changes to `FileInternallyRegistered`. This should not require a DB migration because the fields in the DB are already named differently from their counterparts in the schema itself.
+
+### DCS:
+The DCS subscribes to `FileInternallyRegistered` events from the IFRS to learn about which files are available for download from the Archive. The changes in that event schema, which are described above in the [IFRS section](#ifrs), necessitate database migrations and code updates in the DCS.
+
+#### Migrating existing DCS data
+
+Similar to the IFRS, the DCS should get a new, temporary database collection to preserve the connection between file accession and file ID. This must be done before any of the following changes. Please note that "temporary" here does not mean the table should be dropped at the end of the migration, but that it will be made obsolete and removed after some future development.
+
+The `drs_objects` collection needs the following migration applied:
+- Replace the value of `_id` with the value from `object_id`.
+- Delete `object_id`.
+- Rename `decryption_secret_id` to `secret_id`.
+- Rename `s3_endpoint_alias` to `storage_alias`.
+
+The `dcsPersistedEvents` collection needs the following changes to the `payload` field:
+- Where `type_` == `drs_object_served`:
+  - Replace the value for `file_id` with the value from `target_object_id`
+  - **QUESTION**: *Should we update the actual schema so `s3_endpoint_alias` becomes `storage_alias`?*
+- Where `type_` == `drs_object_registered`:
+  - Replace the value (presently accession) for `file_id` with the corresponding UUID4 value from the temporary table defined above.
+  - `drs_uri` remains untouched...for now.
+    - It's not clear how this will be constructed in the future, but the DCS should only be concerned with files and file IDs rather than file accessions from metadata.
+
+Another note about the DCS migrations is that they should be moved to the init container style. Instead of executing `run_db_migrations()` as part of every entrypoint, the migrations should be run as their own command. 
+
+#### Work to be completed for the DCS
+- [ ] Get the updated `ghga-event-schemas` version
+- [ ] Adapt code for schema updates
+- [ ] Write migrations
+- [ ] Move migrations to own CLI command so they can be run as an init container
+  - [ ] Work with DevOps to get this configured in k8s
+
+## Diagrams:
+
+### Service Diagrams:
+#### Service Map
+![Service Map](./images/service_map.png)
+
```

Comment:
Nice map. The DHFS cleanup job also GETs the inspection results from FIS - that arrow is missing.